### ISSUE TITLE
Can't get the reported performance

### ISSUE THREAD

**Issue by Kyfafyd** [yue-zhongqi/ifsl#3]
Sorry that I can not get the reported performance.
I am running the experiment `SIB/minires_1_ifsl.yaml`, my validation acc is lower than 60%, but reported 68.85%?

Look forward for reply! Thanks!

**Comment by yue-zhongqi** [comment]
I checked the config file and it looked correct.
1. Did you download the mini-ImageNet from the link given? (The dataset should contain .png files of size 84*84)
2. Did you download and use the pre-trained backbone as indicated in the readme.md?

**Comment by Kyfafyd** [comment]
> I checked the config file and it looked correct.
> 
> 1. Did you download the mini-ImageNet from the link given? (The dataset should contain .png files of size 84*84)
> 2. Did you download and use the pre-trained backbone as indicated in the readme.md?

yes, I just follow the readme instruction.

When I am reproducing the experiments, I only meet a problem:
For the following code, this line:
https://github.com/yue-zhongqi/ifsl/blob/eea96618dbaa295be569dafb4707686fedc1b5fe/SIB/PretrainedModel.py#L65
It also include `fc.weight` and `fc.bias` into the` model_params`, in which `fc.weight` with size `[908, 512]` mismatch the size of `fc.weight` (`[64, 512]`) in `model_dict`.
So, considering this, I just removed `fc.weight` and `fc.bias` from` model_params` and load other pretrained params.

I wonder if this is the code wrong? or this is specifically designed?

Thanks for reply and looking forward for further clarification!

**Comment by yue-zhongqi** [comment]
I didn't get this error with this code base. The last fc layer is the pre-trained classifier. It is used for class-wise adjustment and is quite important. For miniImageNet, the fc layer should map 512-dimension feature to 64 classes. Did you load the correct pre-trained model file because the miniImageNet pre-trained model is for 64 classes not 908 classes?

**Comment by Kyfafyd** [comment]
> I didn't get this error with this code base. The last fc layer is the pre-trained classifier. It is used for class-wise adjustment and is quite important. For miniImageNet, the fc layer should map 512-dimension feature to 64 classes. Did you load the correct pre-trained model file because the miniImageNet pre-trained model is for 64 classes not 908 classes?

Oh, I see. Thanks you a lot.
I have double checked that I have loaded the resnet10 pretrained model for miniImageNet, but still get this error...so confusing.
So can you please upload the pretrained models you used in your experiments so I can try once more?

**Comment by yue-zhongqi** [comment]
My server is under maintenance today. Once it is back online, I will help you take a look.

**Comment by Kyfafyd** [comment]
> My server is under maintenance today. Once it is back online, I will help you take a look.

Thanks a lot for keeping attention and sorry for bothering!!!

**Comment by yue-zhongqi** [comment]
I verified that I didn't get this error. I have uploaded the pre-trained weights. Please try this and let me know if it works.
https://drive.google.com/file/d/1wWFQibiLL6yJWNXsHIxioucp0DnxOelP/view?usp=sharing

**Comment by Kyfafyd** [comment]
> I verified that I didn't get this error. I have uploaded the pre-trained weights. Please try this and let me know if it works.
> https://drive.google.com/file/d/1wWFQibiLL6yJWNXsHIxioucp0DnxOelP/view?usp=sharing

Thanks a lot for uploading the models.

The error mentioned before is solved now. However, I got best acc 66.578 +- 0.569%, can't reach reported 68.85.
BTW, I am running `SIB/minires_1_ifsl.yaml`

**Comment by yue-zhongqi** [comment]
Hi I have fixed a bug in network.py. I modified the code (in dni_linear) to test SIB in inductive setting but never changed back. Sorry for the inconvenience caused. I tested on the saved model and below is the test accuracy (I didn't fix random seed so higher than paper).
![image](https://user-images.githubusercontent.com/71916031/101711632-eb487380-3ace-11eb-90ba-4c6603ec89ff.png)

**Comment by Kyfafyd** [comment]
> Hi I have fixed a bug in network.py. I modified the code (in dni_linear) to test SIB in inductive setting but never changed back. Sorry for the inconvenience caused. I tested on the saved model and below is the test accuracy (I didn't fix random seed so higher than paper).
> ![image](https://user-images.githubusercontent.com/71916031/101711632-eb487380-3ace-11eb-90ba-4c6603ec89ff.png)

Thanks for help!
Now I can get 68.65, approximately equal reported 68.85.
Thanks for your nice work again!
